ghostwriter/api/views.py

```diff
@@ -577,6 +581,89 @@ def post(self, request, *args, **kwargs):
+
+    **Parameters**
+
+    ``input``
+        Dictionary containing the evidence ID and optional returnBase64 flag
+    """
+
+    required_inputs = [
+        "id",
+    ]
+
+    def post(self, request, *args, **kwargs):
+        logger.info(
+            "Received evidence download request from %s",
+            request.META.get("HTTP_X_FORWARDED_FOR", request.META.get("REMOTE_ADDR", None)),
+        )
+
+        evidence_id = self.input["id"]
+        return_base64 = self.input.get("returnBase64", False)
+
+        # Get evidence object
+        try:
+            evidence = Evidence.objects.select_related('finding__report__project', 'report__project').get(pk=evidence_id)
+        except Evidence.DoesNotExist:
+            return JsonResponse(
+                utils.generate_hasura_error_payload("Evidence not found", "EvidenceDoesNotExist"),
+                status=HTTPStatus.NOT_FOUND
+            )
+
+        # Check if user has permission to view the evidence
+        project = evidence.finding.report.project if evidence.finding else evidence.report.project
+        if not project.user_can_view(self.user_obj):
+            return JsonResponse(
+            utils.generate_hasura_error_payload("Forbidden - no access to project", "Unauthorized"),
+            status=HTTPStatus.FORBIDDEN
+            )
+
+        # Return response based on request type
+        if return_base64:
+            try:
+                file_data = evidence.document.read()
+                encoded_data = b64encode(file_data).decode('utf-8')
+
+                return JsonResponse({
+                    "id": evidence.id,
+                    "filename": evidence.filename,
+                    "friendlyName": evidence.friendly_name,
+                    "fileBase64": encoded_data,
+                })
+            except FileNotFoundError:
+                logger.error("Evidence file not found during read: %s", evidence.document.path)
+                return JsonResponse(
+                    utils.generate_hasura_error_payload("Evidence file not found on server", "EvidenceFileNotFound"),
+                    status=HTTPStatus.NOT_FOUND
+                )
+            except (IOError, OSError) as e:
+                logger.exception("Failed to read evidence file: %s", e)
+                return JsonResponse(
+                    utils.generate_hasura_error_payload("Failed to read evidence file", "FileReadError"),
+                    status=HTTPStatus.INTERNAL_SERVER_ERROR
+                )
+            except Exception as e:
+                logger.exception("Unexpected error encoding evidence file: %s", e)
+                return JsonResponse(
+                    utils.generate_hasura_error_payload("Failed to encode evidence file", "FileEncodeError"),
+                    status=HTTPStatus.INTERNAL_SERVER_ERROR
+                )
+        else:
+            # Return download URL
+            download_url = request.build_absolute_uri(
```

Comment:
This returns `django:8000` in the URL, but that will only work from within a container. We should return a download link using the configured hostname from the general settings + port. We could also return just the URI for someone to append to their Ghsotwriter instance's protocol, host, and port.